src/power_grid_model/_utils.py

```diff
@@ -148,16 +152,48 @@ def get_batch_size(batch_data: BatchComponentData) -> int:
         indptr = batch_data["indptr"]
         return indptr.size - 1
 
-    data_to_check = (
-        next(iter(cast(DenseBatchColumnarData, batch_data).values()))
-        if is_columnar(batch_data)
-        else cast(DenseBatchArray, batch_data)
-    )
-    if data_to_check.ndim == 1:
+    if not is_columnar(batch_data):
+        sym_array = batch_data
+    else:
+        batch_data = cast(DenseBatchColumnarData, batch_data)
+        if component is None or dataset_type is None:
+            raise ValueError("Cannot deduce batch size for given columnar data without a dataset type or component")
+        sym_attributes, asym_attributes = _get_asym_attributes(dataset_type, component)
+        for attribute, array in batch_data.items():
+            if attribute in sym_attributes:
+                continue
+            if array.ndim == 1:
+                raise TypeError(f"Provided {asym_attributes} is incorrect for the given batch data")
+            if array.ndim == 2:
+                return 1
```

Comment:
The issue stems from the fact that  the dimension of columnar data for asymmetric calculation is 3. Hence it would not be directly known if this data is a single dataset or a batch dataset.
Hence we find if an asymmetric dimension would be present or not from the dataset type.